File: rmt/src/test/java/de/flower/common/ui/ajax/updatebehavior/AjaxUpdateBehaviorTest.java
package de.flower.common.ui.ajax.updatebehavior;

import de.flower.common.logging.Slf4jUtil;
import org.apache.wicket.util.tester.WicketTester;
import org.slf4j.Logger;
import org.testng.annotations.Test;
import static org.testng.Assert.*;

/**
 * @author oblume
 */
public class AjaxUpdateBehaviorTest {

    private final static Logger log = Slf4jUtil.getLogger();

    private WicketTester wicketTester = new WicketTester();

    @Test
    public void testFooBar() {

        wicketTester.startPage(new AjaxUpdateBehaviorTestPage());
        log.info(wicketTester.getLastResponse().getDocument());
        wicketTester.debugComponentTrees();
        wicketTester.clickLink("link1");
        String document = wicketTester.getLastResponse().getDocument();
        log.info(document);
        assertTrue(document.contains(">bar<"));

    }

}

File: rmt/src/test/scala/de/flower/common/ui/ajax/updatebehavior/AjaxUpdateBehaviorTest.scala
package de.flower.common.ui.ajax.updatebehavior

import org.testng.Assert._
import org.testng.annotations.Test
import de.flower.common.test.wicket.AbstractWicketUnitTests

/**
 * @author flowerrrr
 */

class AjaxUpdateBehaviorTest extends AbstractWicketUnitTests {

    @Test
    def testRender() {
        wicketTester.startPage(new AjaxUpdateBehaviorTestPage)
        AbstractWicketUnitTests.log.info(wicketTester.getLastResponse.getDocument)
        wicketTester.debugComponentTrees
        wicketTester.clickLink("link1")
        var document: String = wicketTester.getLastResponse.getDocument
        AbstractWicketUnitTests.log.info(document)
        assertTrue(document.contains(">bar<"))
    }

}